src/view/uniprotkb/components/FeaturesTableView.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { Fragment, useCallback, useState, FC } from 'react';
+import { TemplateResult } from 'lit-html';
+import ProtvistaDatatable from 'protvista-datatable';
+import { UniProtEvidenceTagContent } from '../../../components/UniProtEvidenceTag';
+import { loadWebComponent } from '../../../utils/utils';
+import { ProtvistaFeature, ProcessedFeature } from './FeaturesView';
+import { ProtvistaVariant } from './VariationView';
+import { EvidenceData } from '../../../model/types/EvidenceCodes';
+import { Evidence } from '../../../model/types/modelTypes';
+
+loadWebComponent('protvista-datatable', ProtvistaDatatable);
+
+type FeatureColumns = {
+  [name: string]: {
+    label: string;
+    resolver: (
+      d: ProtvistaFeature & ProtvistaVariant
+    ) => string | number | TemplateResult | TemplateResult[];
+  };
+};
+
+export type FeaturesTableCallback = (
+  evidenceData: EvidenceData,
+  references: Evidence[] | undefined
+) => void;
+
+const FeaturesTableView: FC<{
+  data: ProcessedFeature[] | ProtvistaVariant[];
+  getColumnConfig: (
+    callback: (
+      evidenceData: EvidenceData,
+      references: Evidence[] | undefined
+    ) => void
+  ) => FeatureColumns;
+}> = ({ data, getColumnConfig }) => {
+  const [showEvidenceTagData, setShowEvidenceTagData] = useState(false);
+  const [selectedEvidenceData, setSelectedEvidenceData] = useState();
+  const [selectedReferences, setSelectedReferences] = useState();
+
+  const evidenceTagCallback: FeaturesTableCallback = (
```

Comment:
As `evidenceTagCallback` is only used inside the `useCallback` this could move inside and will prevent redefinition on subsequent passthroughs of `FeaturesTableView`

Reply:
But if it's defined under `evidenceTagCallback` it will be redefined every time the data or column config change? I remember discussing it this morning but now I don't think it would be much of an improvement?

Reply:
I agree, I don't think there is much in it.